Realm/RLMResults.mm

```diff
@@ -376,6 +376,9 @@ - (id)objectAtIndexedSubscript:(NSUInteger)index {
 }
 
 - (id)aggregate:(NSString *)property method:(util::Optional<Mixed> (Results::*)(size_t))method methodName:(NSString *)methodName {
+    if (_results.get_mode() == Results::Mode::Empty) {
```

Comment:
The places where we're currently checking the mode are where calling the functions on an Empty Results would be a precondition violation (e.g. get_query() requires that the Results be non-Empty rather than returning an Optional Query). In this case, calling the aggregate methods on an Empty results is perfectly valid, and is not the thing that needs to be guarded against and is merely correlated with the thing that needs to be guarded against.

Functionally it doesn't matter, but my reaction to seeing a check for Empty was to go look at the Results code  to see why calling the functions on an Empty results was incorrect, while checking `_info` would have made it immediately obvious why the check was being performed.